components/script/canvas_state.rs

```diff
@@ -60,6 +60,39 @@ use crate::dom::paintworkletglobalscope::PaintWorkletGlobalScope;
 use crate::dom::textmetrics::TextMetrics;
 use crate::script_runtime::CanGc;
 
+// Firefox limits width/height to 32767 pixels and Chromium to 65535 pixels,
+// but slows down dramatically before it reaches that limit.
+// We limit by area instead, giving us larger maximum dimensions,
+// in exchange for a smaller maximum canvas size.
+const MAX_CANVAS_AREA: u64 = 32768 * 8192;
+// Max width/height to 65535 in CSS pixels.
+const MAX_CANVAS_SIZE: u64 = 65535;
+
+#[derive(Clone, Debug, JSTraceable, MallocSizeOf)]
+/// Helps observe states of canvas draw target.
+struct DrawTarget {
+    #[no_trace]
+    size: Size2D<u64>,
+    paintable: bool,
+}
+
+impl DrawTarget {
+    fn new(size: Size2D<u64>) -> DrawTarget {
+        DrawTarget {
+            size,
+            paintable: is_supported_canvas_size(size),
+        }
+    }
+
+    fn get_paintable_size(&self) -> Size2D<u64> {
```

Comment:
Getters in Rust usually don't use `get_`: https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter:

```suggestion
    fn paintable_size(&self) -> Size2D<u64> {
```

Reply:
I know about Rust naming policy, but I was confused by existed getter methods names (like get_ipc_renderer, get_canvas_id), so I decided that in Servo are using non Rust naming policy is OK.

Reply:
Servo is very old and predates Rust naming policy, so I think general rule of thumb is that new stuff try to follow it, but existing stuff can keep the old names.